### ISSUE TITLE
java.lang.NullPointerException: Invalid null listener

### ISSUE THREAD

**Issue by fwith** [mcharmas/Android-ReactiveLocation#139]
Hello. We have encountered a problem using this library in our app.

There was a crash. The reason is that onLocationClientReady() is not called and the listener is not initialized.

onUnsubscribed() Commits the code to check whether the Listener is null.

Please confirm. Thank you.

```
Fatal Exception: java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
       at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:59)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
Caused by java.lang.NullPointerException: Invalid null listener
       at com.google.android.gms.common.internal.zzaa.zzb(Unknown Source)
       at com.google.android.gms.location.internal.zzk.zza(Unknown Source)
       at com.google.android.gms.location.internal.zzl.zza(Unknown Source)
       at com.google.android.gms.location.internal.zzd$9.zza(Unknown Source)
       at com.google.android.gms.location.internal.zzd$9.zza(Unknown Source)
       at com.google.android.gms.internal.zzqo$zza.zzb(Unknown Source)
       at com.google.android.gms.internal.zzra.zzd(Unknown Source)
       at com.google.android.gms.internal.zzra.zzb(Unknown Source)
       at com.google.android.gms.internal.zzrf.zzb(Unknown Source)
       at com.google.android.gms.internal.zzrd.zzb(Unknown Source)
       at com.google.android.gms.location.internal.zzd.removeLocationUpdates(Unknown Source)
       at pl.charmas.android.reactivelocation.observables.location.LocationUpdatesObservable.onUnsubscribed(LocationUpdatesObservable.java:45)
       at pl.charmas.android.reactivelocation.observables.BaseObservable$1.call(BaseObservable.java:44)
       at rx.subscriptions.BooleanSubscription.unsubscribe(BooleanSubscription.java:71)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.subscriptions.SequentialSubscription.unsubscribe(SequentialSubscription.java:180)
       at rx.subscriptions.SerialSubscription.unsubscribe(SerialSubscription.java:36)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.subscriptions.SequentialSubscription.unsubscribe(SequentialSubscription.java:180)
       at rx.subscriptions.SerialSubscription.unsubscribe(SerialSubscription.java:36)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.subscriptions.CompositeSubscription.unsubscribeFromAll(CompositeSubscription.java:186)
       at rx.subscriptions.CompositeSubscription.unsubscribe(CompositeSubscription.java:175)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.util.SubscriptionList.unsubscribeFromAll(SubscriptionList.java:136)
       at rx.internal.util.SubscriptionList.unsubscribe(SubscriptionList.java:125)
       at rx.Subscriber.unsubscribe(Subscriber.java:98)
       at rx.internal.operators.OperatorTakeUntil$1.onError(OperatorTakeUntil.java:52)
       at rx.internal.operators.OnSubscribeOnAssembly$OnAssemblySubscriber.onError(OnSubscribeOnAssembly.java:119)
       at rx.internal.operators.OperatorMerge$MergeSubscriber.reportError(OperatorMerge.java:269)
       at rx.internal.operators.OperatorMerge$MergeSubscriber.checkTerminate(OperatorMerge.java:821)
       at rx.internal.operators.OperatorMerge$MergeSubscriber.emitLoop(OperatorMerge.java:582)
       at rx.internal.operators.OperatorMerge$MergeSubscriber.emit(OperatorMerge.java:571)
       at rx.internal.operators.OperatorMerge$InnerSubscriber.onError(OperatorMerge.java:855)
       at rx.internal.operators.OnSubscribeOnAssembly$OnAssemblySubscriber.onError(OnSubscribeOnAssembly.java:119)
       at rx.internal.operators.OperatorSubscribeOn$1$1.onError(OperatorSubscribeOn.java:59)
       at rx.internal.operators.OnSubscribeOnAssembly$OnAssemblySubscriber.onError(OnSubscribeOnAssembly.java:119)
       at rx.internal.operators.OperatorSubscribeOn$1$1.onError(OperatorSubscribeOn.java:59)
       at rx.internal.operators.OnSubscribeOnAssembly$OnAssemblySubscriber.onError(OnSubscribeOnAssembly.java:119)
       at retrofit2.adapter.rxjava.OperatorMapResponseToBodyOrError$1.onError(OperatorMapResponseToBodyOrError.java:52)
       at rx.internal.operators.OnSubscribeOnAssembly$OnAssemblySubscriber.onError(OnSubscribeOnAssembly.java:119)
       at retrofit2.adapter.rxjava.RxJavaCallAdapterFactory$RequestArbiter.request(RxJavaCallAdapterFactory.java:178)
       at rx.internal.operators.OperatorSubscribeOn$1$1$1$1.call(OperatorSubscribeOn.java:85)
       at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:228)
       at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
```

**Comment by mcharmas** [comment]
Strange API behavior but good work. I'll merge it and release new version this week. Thanks.

**Comment by mcharmas** [comment]
Fixed by merging #140
